src/lib/Repository/UiLocationService.php

```diff
@@ -52,6 +63,26 @@ public function loadLocations(ContentInfo $contentInfo)
         return $uiLocations;
     }
 
+    /**
+     * Swaps locations.
+     *
+     * @param int $locationId
+     * @param int $newLocationId
+     */
+    public function swapLocations(int $locationId, int $newLocationId)
+    {
+        $currentLocation = $this->locationService->loadLocation($locationId);
+        $newLocation = $this->locationService->loadLocation($newLocationId);
+
+        $childCount = $this->locationService->getLocationChildCount($currentLocation);
+        $contentType = $this->contentTypeService->loadContentType($newLocation->getContentInfo()->contentTypeId);
+
+        if (!$contentType->isContainer && $childCount) {
+            throw new ForbiddenException('Cannot swap location that has sub items with a location that is not a container');
```

Comment:
Either would work. Although I was thinking of creating a new exception for it to make it more specific something like `LocationSwapForbiddenException`. What are your thoughts on making the exceptions more specific or should we stick to the more generic ones?